pkg/controllers/management/node/utils.go

```diff
@@ -320,3 +321,17 @@ func setEc2ClusterIDTag(data interface{}, clusterID string) {
 		}
 	}
 }
+
+func (m *Lifecycle) getKubeConfig(cluster *v3.Cluster) (*clientcmdapi.Config, error) {
```

Comment:
nit: there maybe be a function for this already but if this one better fits the use case then it's fine.

Reply:
there are but they're elsewhere and need a lot of things inthe controller to make work so i left a local one